### PR TITLE
add query argument parsing for time.Time and *time.Time

### PR DESCRIPTION
**Pull request by Smuerdt**

### CODE REVIEW

File: http/jsonapi/generator/generate_helper.go [hunk 1]
```diff
@@ -23,76 +23,98 @@ func (g *Generator) addGoDoc(typeName, description string) {
...
-			if jsonapi, ok := tags["jsonapi"]; ok { // add hint for jsonapi that time is in iso8601 format
-				tags["jsonapi"] = jsonapi + ",iso8601"
+			if jsonapi, ok := g.tags["jsonapi"]; ok { // add hint for jsonapi that time is in iso8601 format
+				g.tags["jsonapi"] = jsonapi + ",iso8601"
+			} else {
+				addValidator(g.tags, "rfc3339WithoutZone")
+			}
+
+			if g.isParam {
+				g.stmt.Qual("time", "Time") // time.Time can not be nil, so a pointer is needed for omitempty to work
```
Comment (threez) [inline http/jsonapi/generator/generate_helper.go:59]: comment needs to be on line 61

File: http/jsonapi/generator/generate_helper.go [hunk 2]
```diff
@@ -23,76 +23,98 @@ func (g *Generator) addGoDoc(typeName, description string) {
...
+func (g *Generator) goType(stmt *jen.Statement, schema *openapi3.Schema, tags map[string]string) *typeGenerator { // nolint: gocyclo
+	return &typeGenerator{
+		g:      g,
+		stmt:   stmt,
+		schema: schema,
+		tags:   tags,
+	}
+}
+
+type typeGenerator struct {
```
Comment (threez) [inline http/jsonapi/generator/generate_helper.go:35]: a test is missing for the generator, could you please change one of the generator/internal packages and adopt a test so see that it works.

File: http/jsonapi/generator/generate_helper.go (hunk 1 above)
Comment (Smuerdt) [reply http/jsonapi/generator/generate_helper.go]: done

File: http/jsonapi/generator/generate_helper.go (hunk 2 above)
Comment (Smuerdt) [reply http/jsonapi/generator/generate_helper.go:35]: done

**Pull request merged**